### PR TITLE
Added Difficulty Cap To HSM2SignerConfig

### PR DESCRIPTION
**Pull request by kelvinator07**

### CODE REVIEW

**Review by jeremy-then — APPROVED** [review]
LGTM.

**Comment by kelvinator07** [comment]
pipeline:run

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -82,12 +82,12 @@
...
       </component>
-      <component group="co.rsk" name="native" version="1.2.1">
-         <artifact name="native-1.2.1.jar">
-            <sha256 value="01ae15a759c85472f19c9d0dfc33a7d0cbad4f8731a20470b1afdbf4c554ce2e" origin="Generated by Gradle"/>
+      <component group="co.rsk" name="native" version="1.3.0">
+         <artifact name="native-1.3.0.jar">
+            <sha256 value="cf03d2230ae7cf5349b44ffb3f089193aff17d8b8f6071ff562605d1be99228c" origin="Generated by Gradle"/>
          </artifact>
-         <artifact name="native-1.2.1.pom">
-            <sha256 value="6f9c5578df1e4dd58debf5b6693e88cd7a76653908e8e675e9a74ca093231aae" origin="Generated by Gradle"/>
```
Comment (marcos-iov) [inline gradle/verification-metadata.xml:90]: Why these changes?

**Review by marcos-iov — APPROVED** [review]
